Add checkpoint/restore support to the execdriver

boucher opened the pull request:
This PR also includes an individual commit that brings the libcontainer vendor up to date. I'm not sure what the protocol is, but I can separate that out into a separate pull request if desired (but obviously its necessary for this patch).

Commit:
Checkpoint/Restore Support: add exec driver methods

Methods for checkpointing and restoring containers were added to the
native driver.  The LXC driver returns an error message that these
methods are not implemented yet.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	daemon/execdriver/native/create.go
	daemon/execdriver/native/driver.go
	daemon/execdriver/native/init.go

boucher commented on daemon/execdriver/native/driver.go:
```diff
@@ -274,6 +274,144 @@ func (d *driver) Unpause(c *execdriver.Command) error {
+	// so that it'll be owned by this process (Docker daemon) after restore.
+	//
+	// XXX This really belongs to where the Docker daemon starts.
+	if _, _, syserr := syscall.RawSyscall(syscall.SYS_PRCTL, 36, 1, 0); syserr != 0 {
```
@avagin Is this still needed now that we're using criu/swrk?

avagin replied:
No, this isn't needed, because we execute criu with rst_sibling.

duglin commented:
I believe all of the changes under "vendor/*" will need to be done under PRs in those particular projects since docker just copies in their code.

boucher commented:
The changes to vendor are already in libcontainer.

duglin commented:
Then i would think we should pick up the libcontainer update separately. As you suggested in your first comment.

boucher committed:
Update checkpoint/restore support to match docker/master

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher commented:
@crosbymichael ping

LK4D4 commented:
@boucher Something with tags I think, at least for windows.

boucher commented:
Replaced with this PR: https://github.com/docker/docker/pull/13602
